aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -302,10 +302,29 @@ impl AggregatorData {
     /// delta state.
     /// Note: when we say "aggregator instance" here we refer to Rust struct and
     /// not to the Move aggregator.
-    pub fn get_aggregator(&mut self, id: AggregatorID, limit: u128) -> &mut Aggregator {
+    pub fn get_aggregator(
+        &mut self,
+        id: AggregatorID,
+        limit: u128,
+        resolver: &dyn TableResolver,
+        is_aggregator_enabled: bool,
+    ) -> &mut Aggregator {
+        let (state, value) = if is_aggregator_enabled {
```

Comment:
Should we do it only on insertion?

Reply:
I didn't understand. What do you mean doing only on insertion?

Reply:
If aggregator is in the map already, no need to call the resolver. Because we do `self.aggregators.entry(id).or_insert_with(|| ... }`

Reply:
Gotcha. I will first check if the aggregator exists. If new aggregator is to be generated, then I will read the table.

Reply:
Done.